Added section for connecting  to a cluster (#2047)
knative/docs: docs/client/connecting-kn-to-your-cluster.md

---

> title: "Connecting the `kn` Client to your cluster"

Reviewer: I commented about this on slack, but since I had to delete the /client/ index file and readme to make things look okay for the last release, they should be added back in this PR.

linkTitle: "Connecting to your cluster"
weight: 10
type: "docs"
---

> The `kn` Client uses your `kubectl` client configuration, the kubeconfig file, to connect to your cluster.

Reviewer: Looks like there's a weird new line between 8-9. Can you delete that? (May have been my fault from using the "suggestions" feature, so sorry about that!

---
title: "Connecting the `kn` Client to your cluster"
linkTitle: "Connecting to your cluster"
weight: 10
type: "docs"
---

> The `kn` Client uses a Kubernetes client configuraton (kubeconfig) file to connect to your cluster.

Reviewer 2: <!--typo-bot-->
Found potential misspelling(s):
```suggestion
The `kn` Client uses a Kubernetes client configuration (kubeconfig) file to connect to your cluster.
```

> The `kn` Client uses a Kubernetes client configuraton (kubeconfig) file to connect to your cluster.

Reviewer 2: <!--typo-bot-->
Found potential misspelling(s):
```suggestion
The `kn` Client uses a Kubernetes client configuration (kubeconfig) file to connect to your cluster.
```

---
title: "Connecting the `kn` Client to your cluster"
linkTitle: "Connecting to your cluster"
weight: 10
type: "docs"
---

> The `kn` Client requires a `kubeconfig` file to connect to your cluster.

Reviewer 3: Are there any requirements that need to be listed here? 

Its not clear what the task is here? Does the user need to take action?

Author: It's something that's a bigger Kubernetes concept, I've added a link for more details below: https://github.com/knative/docs/pull/2047/files/f896f7269dc7e3fd59e11f1c9050a3f853234786#diff-ad04a060c15f7df4940a8878b45d1a63R14

AFAIK it's something they should have by default?
I'm not sure if we need to include more than that? @rhuss can you advise if there's some specific action we need to call out to make this clearer?

Reviewer 3: I think this needs to be revised for clarity. ie "requires" sounds like an action item for the user (and there is nothing here that is requesting an action). if this is just the conceptual overview (and not a task as the title indicates ie "connecting to the...") then the title and filename should be changed too. 

Example for this particular line: "the kn client uses the kubeconfif file of your cluster to establish a connection when .....(missing detail: say why kn connects here)"

On the other hand, the user does need to do something to that kubeconfig file in order to connect, we need to state that here. For example, is there a particular element in the kubeconfig file that is "required", if yes, then we should state to look for that ..."ensure the...".

Thanks for all the help with fleshing out the missing info @abrennan89 !
I'm being temporarily assigned to another project so i'll let @samodell take it from here.

Author: @rhuss re Richie's question, **why** does `kn` need to connect to a cluster?

> The `kn` Client requires a `kubeconfig` file to connect to your cluster.

Reviewer 4: I wouldn't  use `kubeconfig`  as a literal as there is typically no file called `kubeconfig` (instead it's a `config` file in the `~/.kube`. What about's just calling it "Kubernetes client configuraton" instead of "`kubeconfig` file". Or at least render kubeconfig in normal font (as I see that Kubernetes over there also uses the terminology "kubeconfig file")

> The `kn` Client requires a `kubeconfig` file to connect to your cluster.

Reviewer 4: ```suggestion
The `kn` client requires a kubeconfig file to connect to your cluster.
```

> `kn` will pick up your `kubeconfig` file in the default location of `$HOME/.kube/config`.

Reviewer 4: ```suggestion
`kn` will pick up your kubeconfig file in the default location of `$HOME/.kube/config`.
```

> You can specify an alternate `kubeconfig` connection file by using `--kubeconfig`, or the `$KUBECONFIG` environment variable, for any command.

Reviewer 4: ```suggestion
You can specify an alternate kubeconfig file by using `--kubeconfig`, or setting the `$KUBECONFIG` environment variable, for any `kn` command.
```

<!--example of this?-->

> For more information about `kubeconfig` files. see [Organizing Cluster Access Using kubeconfig Files](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/).

Reviewer 4: ```suggestion
For more information about kubeconfig files. see [Organizing Cluster Access Using kubeconfig Files](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/).
```

> ## Using `kubeconfig` files with your platform

Reviewer 4: ```suggestion
## Using kubeconfig files with your platform
```

> Instructions for using `kubeconfig` files are available for the following platforms:

Reviewer 4: ```suggestion
Instructions for using kubeconfig files are available for the following platforms:
```
